public/services/data_source_service.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { BehaviorSubject, PartialObserver, Subscription } from 'rxjs';
+import type { IUiSettingsClient } from '../../../../src/core/public';
+import type { DataSourceOption } from '../../../../src/plugins/data_source_management/public/components/data_source_menu/types';
+import type { DataSourceManagementPluginSetup } from '../types';
+
+export enum DataSourceIdFrom {
+  UiSettings,
+  DataSourceSelection,
+  Customized,
+}
+
+const getDSMDataSourceSelectionOption = (dataSourceSelection: Map<string, DataSourceOption[]>) => {
```

Comment:
The function name seems not comply with what it is doing. What about `getSingleSelectedDataSourceOption`?